go/vt/sqlparser/ast_rewriting.go

```diff
@@ -386,21 +400,36 @@ func (er *astRewriter) rewrite(cursor *Cursor) bool {
 			cursor.Replace(inner)
 		}
 	case *AliasedTableExpr:
-		if !SystemSchema(er.keyspace) {
-			break
-		}
 		aliasTableName, ok := node.Expr.(TableName)
 		if !ok {
-			return true
+			break
 		}
 		// Qualifier should not be added to dual table
-		if aliasTableName.Name.String() == "dual" {
+		tblName := aliasTableName.Name.String()
+		if tblName == "dual" {
+			break
+		}
+		if SystemSchema(er.keyspace) {
+			if er.keyspace != "" && aliasTableName.Qualifier.IsEmpty() {
+				aliasTableName.Qualifier = NewIdentifierCS(er.keyspace)
+				node.Expr = aliasTableName
+				cursor.Replace(node)
+			}
+			break
+		}
```

Comment:
`er.keyspace != ""` this should be true always as the earlier check `SystemSchema(er.keyspace)` is true.
other question is why this rewriting is needed here?

Reply:
👍